### DISCUSSION TITLE
Install Failed on Raspberry Pi 4 CM4 rev 1.1

### DISCUSSION THREAD

**Discussion by dinkenstein** [marcone/teslausb#858] [Q&A]
This Failed to install on Pi compute module 4, I thought I'd give it a try as the hardware is essentially the same as Pi4 running on same Pi OS. Is there anyway to force install and ignore the unsupported hardware check? Thank you Thank you!

Downloaded /tmp/verify-configuration.sh ...
/tmp/verify-configuration.sh: line 33: warning: command substitution: ignored null byte in input
STOP: unsupported hardware: 'Raspberry Pi Compute Module 4 Rev 1.1'
(only Pi Zero W, Pi 4, and Pi 5 have the necessary hardware to run teslausb)

**Comment by Arthur-Philip-Dent** [comment]
> is essentially the same as Pi4 running on same Pi OS

Not really... it depends on the base board you plug the CM4 into. 
Can your base board handle the OTG/gadget-device properly? 

I had 3 different bases from Waveshare: 
- Nano Base Board (A) -> works (gadget is on the USB-A connector)
- CM42P4-Expander -> works (gadget is on any of the 4 USB-A connectors)
- CM4-IO-Base-Board-C -> needs soldering to have a the gadget-device working properly through the USB-C, where also the power comes from. 

I did my own repo on github to test out and it worked for the "2,5 boards" I tinkered around with.

**Reply by dinkenstein** [reply]
I have Nano base board (B) version with LAN port, so technically it should work. 
1. Did you connect to Tesla with a USA A to USB A cable?
2. How did you bypass the unsupported hw check? Is it available to public? I tried it a unsupported OrangePi as well, everything installed and there was no hardware block, thought I'd have better luck with a CM4.

**Comment by Arthur-Philip-Dent** [comment]
Marcone porbably won't support the CM4 for exactly this reason, since it depends on the base, if it works or not. We could make some suggestions to identify the workable combinations to add to his repo to keep maintenance as low as possible for him.

**Reply by marcone** [reply]
Is it possible to tell which base the CM is plugged in to? The supported hardware check was originally added because people were trying to use their old Raspberry Pi 2 and 3 to run TeslaUSB, and then wondering why that didn't work. Perhaps changing the current check to specifically reject the 2 and 3 would be a decent solution, the assumption being that all other and future Raspberry Pi support OTG.

**Reply by Arthur-Philip-Dent** [reply]
Tested base working out of the box with each 2/4/8GB CM4 module, WiFi, NoEmmc
- Waveshare Nano-Base-A
- Waveshare CM4-To-Pi4B-Adaptor

Would appreciate suggested solution for HW-Check. 

Haven’t checked the eeprom_data for differences. They would require i2detect/i2c-tools.

**Reply by dinkenstein** [reply]
Or add CM4s to the safe-list?  With a warning message: at your own risk as your hardware may not be supported.

Eitherway works fine and really appreciate your excellent work!!

**Comment by Arthur-Philip-Dent** [comment]
Definitely the easy solution. Thats what @marcone suggested. I would go for it. 
I won’t have an answer before Sunday anyways if I can look it up in the eeprom (or devicetree), or not.
